### PR TITLE
FIX: #3262 custom files issues

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/ensemblvep/vep/meta.yml [hunk 1]
```diff
@@ -27,7 +27,6 @@ input:
       type: file
       description: |
         extra sample-specific files to be used with the `-custom` flag
```
Comment (maxulysse) [inline modules/nf-core/ensemblvep/vep/meta.yml:29]:
```suggestion
        extra sample-specific files to be used with the `--custom` flag to be configured with text.args
```

File: subworkflows/nf-core/vcf_annotate_ensemblvep/main.nf [hunk 2]
```diff
@@ -7,26 +7,27 @@ include { TABIX_TABIX    } from '../../../modules/nf-core/tabix/tabix/main'
...
-    } else {
-        ch_vep_input = ch_vcf.map { it + [[]] }
-    }
+    ch_vcf.view()
+    // ch_fasta.view()
+    // ch_cache.view()
+    // ch_extra_files.view()
+    println val_genome
+    println val_species
+    println val_cache_version
```
Comment (maxulysse) [inline subworkflows/nf-core/vcf_annotate_ensemblvep/main.nf:28]:
```suggestion
```

File: subworkflows/nf-core/vcf_annotate_ensemblvep/main.nf [hunk 3]
```diff
@@ -7,26 +7,27 @@ include { TABIX_TABIX    } from '../../../modules/nf-core/tabix/tabix/main'
 
 workflow VCF_ANNOTATE_ENSEMBLVEP {
     take:
-    ch_vcf                      // channel: [ val(meta), path(vcf) ]
+    ch_vcf                      // channel: [ val(meta), path(vcf), [path(custom_extra_files)] ]
```
Comment (maxulysse) [inline subworkflows/nf-core/vcf_annotate_ensemblvep/main.nf:10]:
```suggestion
    ch_vcf                      // channel: [ val(meta), path(vcf), [path(custom_file1), path(custom_file2)...]]
```

File: modules/nf-core/ensemblvep/vep/meta.yml [hunk 4]
```diff
@@ -26,8 +26,7 @@ input:
   - custom_extra_files:
       type: file
       description: |
-        extra sample-specific files to be used with the `-custom` flag
-        => add this yourself in the `args`
+        extra sample-specific files to be used with the `--custom` flag to be configured with text.args
```
Comment (maxulysse) [inline modules/nf-core/ensemblvep/vep/meta.yml:29]:
```suggestion
        extra sample-specific files to be used with the `--custom` flag to be configured with ext.args
```

File: subworkflows/nf-core/vcf_annotate_ensemblvep/main.nf [hunk 5]
```diff
@@ -7,26 +7,19 @@ include { TABIX_TABIX    } from '../../../modules/nf-core/tabix/tabix/main'
 
 workflow VCF_ANNOTATE_ENSEMBLVEP {
     take:
-    ch_vcf                      // channel: [ val(meta), path(vcf) ]
+    ch_vcf                      // channel: [ val(meta), path(vcf), [path(custom_file1), path(custom_file2)...]]
```
Comment (maxulysse) [inline subworkflows/nf-core/vcf_annotate_ensemblvep/main.nf:10]:
```suggestion
    ch_vcf                      // channel: [ val(meta), path(vcf), [path(custom_file1), path(custom_file2)... (optionnal)]]
```

File: subworkflows/nf-core/vcf_annotate_ensemblvep/meta.yml [hunk 6]
```diff
@@ -11,7 +11,7 @@ input:
   - ch_vcf:
       description: |
         vcf file to annotate
-        Structure: [ val(meta), path(vcf) ]
+        Structure: [ val(meta), path(vcf), [path(custom_file1), path(custom_file2)...] ]
```
Comment (maxulysse) [inline subworkflows/nf-core/vcf_annotate_ensemblvep/meta.yml:14]:
```suggestion
        Structure: [ val(meta), path(vcf), [path(custom_file1), path(custom_file2)... (optionnal)] ]
```

File: tests/modules/nf-core/ensemblvep/vep/main.nf [hunk 7]
```diff
@@ -106,5 +107,21 @@ workflow test_ensemblvep_vep_no_fasta {
         []
     ]
 
-    ENSEMBLVEP_VEP_DEFAULT ( input, "WBcel235", "caenorhabditis_elegans", "108", [[],[]], [[],[]], [] )
+    ENSEMBLVEP_VEP_DEFAULT ( input, "WBcel235", "caenorhabditis_elegans", "108", [], [[],[]], [] )
```
Comment (maxulysse) [inline tests/modules/nf-core/ensemblvep/vep/main.nf:110]:
```suggestion
    ENSEMBLVEP_VEP_DEFAULT ( input, "WBcel235", "caenorhabditis_elegans", "108", [], [[], []], [] )
```

**Comment by maxulysse** [comment]
All tests are failing in #3262, hopefully this PR fixes all that

**Review by nvnieuwk — APPROVED** [review]
Awesome this is way better! Thank you!

**Pull request merged**